### PR TITLE
[Merged by Bors] - refactor: deprecate `StarAlgHomClass`  and similar as they just duplicate `StarHomClass`

### PR DESCRIPTION
**Pull request by j-loreaux**
When `FunLike` (and `EquivLike`) was unbundled from all the hom classes, it was necessary to make `StarAlgHomClass` (and similarly `NonUnitalStarAlgHomClass` and `StarAlgEquivClass`) and its un-starred counterparts an argument instead of extending that class. The net effect of this was that these classes became duplicates of `StarHomClass` with stricter type class assumptions.

This PR deprecates all of these classes in favor of `StarHomClass` alone.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
!bench

File: Mathlib/Algebra/Star/StarAlgHom.lean [hunk 1]
```diff
@@ -303,35 +297,16 @@ structure StarAlgHom (R A B : Type*) [CommSemiring R] [Semiring A] [Algebra R A]
 by forgetting the interaction with the star operation. -/
 add_decl_doc StarAlgHom.toAlgHom
 
-/-- `StarAlgHomClass F R A B` states that `F` is a type of ⋆-algebra homomorphisms.
-
-You should also extend this typeclass when you extend `StarAlgHom`. -/
-class StarAlgHomClass (F : Type*) (R A B : outParam Type*)
```
Comment (eric-wieser) [inline Mathlib/Algebra/Star/StarAlgHom.lean:309]: I wonder if we should keep this around but mark it `@[deprecated]`?

**Comment by j-loreaux** [comment]
!bench

File: Mathlib/Algebra/Star/StarAlgHom.lean (hunk 1 above)
Comment (j-loreaux) [reply Mathlib/Algebra/Star/StarAlgHom.lean:309]: done

**Comment by dupuisf** [comment]
Thanks!

bors r+

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**